chain/manager.go

```diff
@@ -806,15 +808,17 @@ func (m *Manager) TransactionsForPartialBlock(missing []types.Hash256) (txns []t
 	for _, txn := range m.txpool.txns {
 		if h := txn.FullHash(); want[h] {
 			txns = append(txns, txn)
-			if delete(want, h); len(want) == 0 {
+			delete(want, h)
+			if len(want) == 0 {
```

Comment:
I like this pattern:
```go
if change x; check something about x {
    // ...
}
```
e.g.
```go
if i++; i > 10 {
```
or:
```go
if peers = append(peers, p); len(p) > 10 {
```
I think it follows naturally from the `if err := foo(); err != nil` pattern. It gets a little weird with `delete` because `delete` doesn't return a value, but the principle is the same.

This isn't one of the hills that I care enough to die on, though. :P Just wanted to explain the reasoning here.